dalm/pipelines/reading_comprehension_pipeline.py

```diff
@@ -0,0 +1,383 @@
+
+    dataset.save_to_disk(output_dataset_name)
+
+    train_generator(
+        model_name=model_name,
+        dataset_name=output_dataset_name,
+        num_train_epochs=num_train_epochs,
+        split=split,
+        size_valid_set=size_valid_set,
+        streaming=streaming,
+        shuffle_buffer=shuffle_buffer,
+        seq_length=seq_length,
+        num_workers=num_workers,
+        eval_steps=eval_steps,
+        logging_steps=logging_steps,
+        per_device_train_batch_size=per_device_train_batch_size,
+        per_device_eval_batch_size=per_device_eval_batch_size,
+        gradient_accumulation_steps=gradient_accumulation_steps,
+        gradient_checkpointing=gradient_checkpointing,
+        group_by_length=group_by_length,
+        packing=packing,
+        lora_alpha=lora_alpha,
+        lora_dropout=lora_dropout,
+        lora_r=lora_r,
+        learning_rate=learning_rate,
+        lr_scheduler_type=lr_scheduler_type,
+        num_warmup_steps=num_warmup_steps,
+        weight_decay=weight_decay,
+        optimizer_type=optimizer_type,
+        output_dir=model_output_dir,
+        neftune_noise_alpha=neftune_noise_alpha,
+        log_with=log_with,
+        local_dataset=True,
+        validation_split=validation_split,
+        run_name=run_name,
+    )
+
+    if generation_state_file:
+        os.remove(generation_state_file)
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--model_name", type=str, default="HuggingFaceH4/zephyr-7b-alpha", help="name of the model to be trained"
+    )
+    parser.add_argument(
+        "--output_dataset_name",
+        type=str,
+        required=True,
+        help="name of the dataset that the generated data will be saved to",
+    )
+    parser.add_argument(
+        "--comprehension_type",
+        type=SynthMode,
+        default=SynthMode.BOTH,
+        choices=list(SynthMode),
+        help="type of comprehension to be generated",
+    )
+    parser.add_argument(
+        "--llm_synth_model_name",
+        type=str,
+        default="HuggingFaceH4/zephyr-7b-beta",
+        help="name of the model to be used for LLM based generation",
+    )
+    parser.add_argument(
+        "--llm_synth_model_context_length", type=int, default=4096, help="context length to calulcate the chunk size"
+    )
+    parser.add_argument(
+        "--llm_dataset_output_path",
+        type=str,
+        default="llm_dataset",
+        help="path to save the generated LLM based dataset",
+    )
+    parser.add_argument(
+        "--general_spm_path",
+        type=str,
+        default="./resources/general.spm",
+        help="path to the general tokenizer (needed for regex based generation)",
+    )
+    parser.add_argument(
+        "--domain_spm_path",
+        type=str,
+        default=None,
+        help="path to the domain tokenizer (needed for regex based generation)",
```

Comment:
```suggestion
        help="path to the domain tokenizer used for regex based generation.  If None is provided (default), it will be automatically generated.",
```